package/MDAnalysis/core/AtomGroup.py

```diff
@@ -4715,6 +4726,47 @@ def _matches_unpickling(self, anchor_name, n_atoms, fname, trajname):
         else:
             return False
 
+    def transfer_to_memory(self, frame_interval=1):
+        """Replace the current trajectory reader object with one of type
+        :class:`MDAnalysis.coordinates.memory.MemoryReader` to support in-place
+        editing of coordinates.
+
+        :Arguments:
+             *frame_interval*
+                 Read in every nth frame.
+         """
+
+        from ..coordinates.memory import MemoryReader
+
+        if self.trajectory.format != "array":
+
+            # Try to extract coordinates using Timeseries object
+            # This is significantly faster, but only implemented for certain
+            # trajectory file formats
+            try:
+                coordinates = self.universe.trajectory.timeseries(
+                    self.atoms, format='afc', skip=frame_interval)
+
+            # if the Timeseries extraction fails,
+            # fall back to a slower approach
+            except AttributeError:
+                coordinates = np.zeros(
+                    tuple([self.universe.trajectory.n_frames/frame_interval]) +
+                    self.atoms.coordinates().shape)
+
+                k = 0
+                for i, time_step in enumerate(self.universe.trajectory):
```

Comment:
Oh, and for the record, I had to change your suggestion above to:

```python
                coordinates = \
                    np.array([ts.positions for ts in
                            self.trajectory[frame_interval-1::frame_interval]])
                coordinates = coordinates.swapaxes(0, 1)
```
since `__item__` in trajectory returns a timestep object.